Add xata branch resolution from API

SferaDev opened the pull request:
- Add branch resolution to the SDK
  - Read from API linked branch and use it during infer
- Add ``link`` and ``unlink`` commands to ``branches`` CLI
  - Add/Remove linked branches
- Update ``create`` command in ``branches`` CLI
  - Link branches on create
  - Resolve base git branch using ``from`` linked xata branch
- Update ``list`` command in ``branches`` CLI
  - Show linked branches in listing

gimenete commented on cli/src/commands/branches/create.ts:
```diff
@@ -57,12 +66,12 @@ export default class BranchesCreate extends BaseCommand {
 
     const result = await xata.branches.createBranch(workspace, database, branch, from);
+    if (useGit) {
+      await xata.databases.addGitBranchesEntry(workspace, database, { xataBranch: branch, gitBranch: branch });
```
I think we can omit this step since the git branch and the xata branch have the same name.

gimenete commented on packages/client/src/util/config.ts:
```diff
@@ -24,30 +24,47 @@ export async function getCurrentBranchName(options?: BranchResolutionOptions): P
 }
 
 export async function getCurrentBranchDetails(options?: BranchResolutionOptions) {
-  const env = getBranchByEnvVariable();
-  if (env) return getDatabaseBranch(env, options);
```
🤔 I think we should keep this. Any reason to delete it?

gimenete reviewed (commented):
It's lookin great! I left a few comments.

SferaDev commented on packages/client/src/util/config.ts:
```diff
@@ -24,30 +24,47 @@ export async function getCurrentBranchName(options?: BranchResolutionOptions): P
+      fetchImpl: getFetchImplementation(options?.fetchImpl),
+      workspacesApiUrl: `${protocol}//${host}`,
+      pathParams: { dbName, workspace },
+      queryParams: { gitBranch, fallbackBranch: defaultBranch }
```
Should it also be the ``defaultBranch``?

SferaDev commented:
@gimenete Removed the ``defaultBranch``, added a check for the ``XATA_BRANCH`` to exist, added the ``XATA_FALLBACK_BRANCH`` env variable.

Also removed the linking on ``create`` and improved the ``list`` command output:

![image](https://user-images.githubusercontent.com/2181866/173628716-3d3edece-c6cb-4c73-84d4-c146c8c5c339.png)

gimenete reviewed (approved):
LGTM!

It's a bit werid that the unlink command doesn't ask for a xata branch. But I guess that's how the endpoint works: it deletes all mappings for a given git branch.
